src/main/java/com/exadel/discount/service/impl/VendorLocationServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.exadel.discount.service.impl;
+
+import com.exadel.discount.dto.VendorLocationDTO;
+import com.exadel.discount.entity.Vendor;
+import com.exadel.discount.entity.VendorLocation;
+import com.exadel.discount.exception.NotFoundException;
+import com.exadel.discount.mapper.VendorLocationMapper;
+import com.exadel.discount.repository.VendorLocationRepository;
+import com.exadel.discount.service.VendorLocationService;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class VendorLocationServiceImpl implements VendorLocationService {
+
+    private final VendorLocationRepository vendorLocationRepository;
+    private final VendorLocationMapper vendorLocationMapper;
+
+    @Override
+    public VendorLocationDTO save(VendorLocation vendorLocation, UUID vendorId) {
+        log.debug("Saving new VendorLocation");
+        Vendor vendor = new Vendor();
+        vendor.setId(vendorId);
```

Comment:
I think here we should get Vendor by id from vendorRepository and then set it to vendorLocation

Reply:
Its work